### PR TITLE
Remember guest names

### LINKED ISSUE #552: Guest username is not remembered to host after guest refreshes
## Steps to reproduce
1. Host a call
2. Have some guests, with names put in
3. One of them refreshes

### Expected behaviour
Comes back and is shown with name and colored placeholder avatar as before.

### Actual behaviour
Is shown as Guest with grey placeholder avatar.

See screenshot (the same person had a name set shortly before as you can see in https://github.com/nextcloud/spreed/issues/551#issue-287195223)
screenshot from 2018-01-09 19-21-04

Happened on our cloud.nc

@nickvergessen @Ivansss @danxuliu

### PR DESCRIPTION
**Pull request by Ivansss**
Now, the guest name is sent every time the connection is established.
Also guest names are stored by every participant in a (session -> guest name) table.
So screenshare's name indicator can be updated properly.

Fix #552

### CODE REVIEW

**Comment by jancborchardt** [comment]
@Ivansss if they are saved in a table, does it also fix this issue? »Guest usernames and avatar placeholders are not updated in sidebar (Chat, Participants)« #551

**Comment by Ivansss** [comment]
Not really. It could work as a workaround for the sidebar. But for chat, this info should be saved in the database, so guest names could be kept in chat history.

**Comment by nickvergessen** [comment]
Table is misused here, he ment a js array ;)

File: js/webrtc.js [hunk 1]
```diff
@@ -881,7 +879,7 @@ var spreedPeerConnectionTable = [];
 
 			var screenNameIndicator = $(screen).find('.nameIndicator');
 
-			if (data.name.length === 0) {
+			if (!data.name || data.name.length === 0) {
```
Comment (nickvergessen) [inline js/webrtc.js]: get rid of ` || data.name.length === 0` all strings that match it, also match `!data.name`

**Pull request merged**